core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/EvaluationStrategy.java

```diff
@@ -95,6 +96,16 @@ TupleExpr optimize(TupleExpr expr, EvaluationStatistics evaluationStatistics,
 	Value evaluate(ValueExpr expr, BindingSet bindings)
 			throws ValueExprEvaluationException, QueryEvaluationException;
 
+	/**
+	 * Gets the value of this expression.
+	 *
+	 * @param expr
+	 * @param bindings The variables bindings to use for evaluating the expression, if applicable.
+	 * @return The Value that this expression evaluates to, or <tt>null</tt> if the expression could not be evaluated.
+	 */
+	Value evaluate(Var expr, BindingSet bindings)
+			throws ValueExprEvaluationException, QueryEvaluationException;
+
```

Comment:
Thought adding this would cut down on the number of instanceof calls we were making, but doesn't actually seem to make a difference.